File: app/src/main/java/io/github/coffeegerm/materiallogbook/ui/activity/SettingsActivity.java
package io.github.coffeegerm.materiallogbook.ui.activity;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.CompoundButton;
import android.widget.LinearLayout;
import android.widget.Switch;
import android.widget.TextView;
import android.widget.Toast;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.github.coffeegerm.materiallogbook.R;
import io.github.coffeegerm.materiallogbook.model.EntryItem;
import io.realm.Realm;

/**
 * Created by David Yarzebinski on 6/25/2017.
 * <p>
 * Activity for changing and showing chosen settings of Material Logbook.
 */

public class SettingsActivity extends AppCompatActivity {

    private static final String TAG = "SettingsActivity";
    @BindView(R.id.btn_delete_all)
    TextView deleteAllEntries;
    @BindView(R.id.toggle_dark_mode)
    Switch toggleDarkMode;
    @BindView(R.id.setting_toolbar)
    Toolbar settingsToolbar;
    @BindView(R.id.treatment_section)
    LinearLayout treatmentSection;
    private Realm realm;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (MainActivity.sharedPreferences.getBoolean(MainActivity.PREF_DARK_MODE, false)) {
            setTheme(R.style.AppTheme_Dark);
        }
        setContentView(R.layout.activity_settings);
        ButterKnife.bind(this);
        initView();
    }

    public void initView() {
        realm = Realm.getDefaultInstance();
        setupToolbar();

        toggleDarkMode.setChecked(MainActivity.sharedPreferences.getBoolean(MainActivity.PREF_DARK_MODE, false));
        toggleDarkMode.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                MainActivity.sharedPreferences.edit().putBoolean(MainActivity.PREF_DARK_MODE, isChecked).apply();
                recreate();
            }
        });

        treatmentSection.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(getApplicationContext(), SettingsTreatmentActivity.class));
            }
        });

        deleteAllEntries.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                final AlertDialog.Builder builder;

                // Sets theme based on VERSION_CODE
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)
                    builder = new AlertDialog.Builder(SettingsActivity.this, android.R.style.Theme_Material_Dialog_NoActionBar);
                else builder = new AlertDialog.Builder(SettingsActivity.this);

                builder.setTitle("Delete all entries")
                        .setMessage("Are you sure you want to delete all entries?")
                        .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
                            public void onClick(DialogInterface dialog, int which) {
                                // continue with delete
                                realm.executeTransaction(new Realm.Transaction() {
                                    @Override
                                    public void execute(Realm realm) {
                                        realm.delete(EntryItem.class);
                                    }
                                });
                                Toast.makeText(SettingsActivity.this, R.string.all_deleted, Toast.LENGTH_SHORT).show();
                            }
                        })
                        .setNegativeButton(android.R.string.no, new DialogInterface.OnClickListener() {
                            public void onClick(DialogInterface dialog, int which) {
                                // do nothing
                                dialog.dismiss();
                            }
                        })
                        .setIcon(R.drawable.ic_trash)
                        .show();
            }
        });
    }

    public void setupToolbar() {
        setSupportActionBar(settingsToolbar);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setDisplayShowHomeEnabled(true);
            getSupportActionBar().setTitle(R.string.settings);
        }
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return super.onSupportNavigateUp();
    }

    @Override
    public void onDestroy() {
        realm.close();
        super.onDestroy();
    }
}

File: app/src/main/java/io/github/coffeegerm/materiallogbook/ui/settings/SettingsActivity.kt
/*
 * Copyright 2018 Coffee and Cream Studios
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.github.coffeegerm.materiallogbook.ui.settings

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import io.github.coffeegerm.materiallogbook.MaterialLogbook.Companion.syringe
import io.github.coffeegerm.materiallogbook.R
import io.github.coffeegerm.materiallogbook.ui.settings.children.SettingsDataActivity
import io.github.coffeegerm.materiallogbook.ui.settings.children.SettingsTreatmentActivity
import io.github.coffeegerm.materiallogbook.utils.Constants
import io.github.coffeegerm.materiallogbook.utils.Constants.PREF_DARK_MODE
import kotlinx.android.synthetic.main.activity_settings.*
import javax.inject.Inject

class SettingsActivity : AppCompatActivity() {
  
  @Inject
  lateinit var sharedPreferences: SharedPreferences
  
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    syringe.inject(this)
    if (sharedPreferences.getBoolean(Constants.PREF_DARK_MODE, false)) setTheme(R.style.AppTheme_Dark)
    setContentView(R.layout.activity_settings)
    initView()
  }
  
  private fun initView() {
    setupToolbar()
    
    toggle_dark_mode.isChecked = sharedPreferences.getBoolean(Constants.PREF_DARK_MODE, false)
    toggle_dark_mode.setOnCheckedChangeListener({ _, isChecked -> sharedPreferences.edit().putBoolean(PREF_DARK_MODE, isChecked).apply() })
    
    military_time_switch.isChecked = sharedPreferences.getBoolean(Constants.MILITARY_TIME, false)
    military_time_switch.setOnCheckedChangeListener { _, isChecked -> sharedPreferences.edit().putBoolean(Constants.MILITARY_TIME, isChecked).apply() }
    
    treatment_section.setOnClickListener({ startActivity(Intent(applicationContext, SettingsTreatmentActivity::class.java)) })
    
    data_section.setOnClickListener { startActivity(Intent(applicationContext, SettingsDataActivity::class.java)) }
  }
  
  private fun setupToolbar() {
    setSupportActionBar(setting_toolbar)
    if (supportActionBar != null) {
      supportActionBar!!.setDisplayHomeAsUpEnabled(true)
      supportActionBar!!.setDisplayShowHomeEnabled(true)
      supportActionBar!!.setTitle(R.string.settings)
    }
  }
  
  override fun onSupportNavigateUp(): Boolean {
    onBackPressed()
    return super.onSupportNavigateUp()
  }
}